### PR TITLE
Modernize code (remove deprecated, backward compatibility)

### PR DESCRIPTION
**Pull request by jonathanmetzman**
Replace disutils.spawn.find_executable with shutil.which.
distutils is deprecated.
Also stop using `pipes.quote`, we no longer need to support <python3.3.
Use f-strings where possible.

### CODE REVIEW

**Review by hogo6002 — APPROVED** [review]

**Review by oliverchang — APPROVED** [review]

**Review by oliverchang — APPROVED** [review]

**Review by Alan32Liu — APPROVED** [review]

**Comment by Alan32Liu** [comment]
Would it also make sense to include the following in this PR:
1. Change `src/local/butler/common.py:427` from `dir_util.copy_tree(...)` to `shutil.copytree(...)`?
2. Remove `from distutils import dir_util` in `src/local/butler/common.py:21` (Pacakge `shutil` is already imported).

**Comment by jonathanmetzman** [comment]
PTAL again as I've changed quite a bit.

File: src/local/butler/common.py [hunk 1]
```diff
@@ -292,26 +286,22 @@ def _install_platform_pip(requirements_path, target_path, platform_name):
...
+        f'{temp_dir}',
         exit_on_error=False)
 
     if return_code != 0:
-      print('Did not find package for platform: ' + pip_platform)
+      print(f'Did not find package for platform: {pip_platform}')
       continue
 
-    execute('unzip -o -d %s \'%s/*.whl\'' % (target_path, temp_dir))
+    execute(f'unzip -o -d {target_path} \'{temp_dir}/*.whl\'')
```
Comment (Alan32Liu) [inline src/local/butler/common.py:298]: nit: Same question about single/double quotes.

**Review by Alan32Liu — APPROVED** [review]

File: src/local/butler/common.py [hunk 2]
```diff
@@ -150,17 +144,17 @@ def _print(s):
     if print_output:
       print(s)
 
-  print_string = 'Running: %s' % command
+  print_string = f'Running: {command}'
   if cwd:
-    print_string += " (cwd='%s')" % cwd
+    print_string += f' (cwd=\'{cwd}\')'
```
[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply src/local/butler/common.py:149]: Done.

File: src/local/butler/common.py (hunk 1 above)
Comment (jonathanmetzman) [reply src/local/butler/common.py:298]: Done.

**Pull request merged**